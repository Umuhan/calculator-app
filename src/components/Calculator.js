import React, { useState } from "react";
import calculate from "../logic/calculate";

const Calculator = () => {
  const [count, setCount] = useState({
    total: 0,
    operation: null,
    next: null,
  });

  const clickHandler = (button) => {
    const output = calculate(count, button);
    setCount(output);
  };

  // eslint-disable-next-line
  const Button = ({ label, className }) => (
    <button
      type="button"
      onClick={() => clickHandler(label)}
      className={className}
    >
      {label}
    </button>
  );

  return (
    <div>
      <h1 style={{ textAlign: "left" }} className="math-magic">
        Math Calculator
      </h1>
      <h4 className="heading">Lets do some math!</h4>

      <main className="container">
        <section className="calculator">
          <header className="header calc">
            {count.total} {count.operation} {count.next}
          </header>
          <Button label="AC" />
          <Button label="+/-" />
          <Button label="%" />
          <Button label="รท" className="color" />
          <Button label="7" />
          <Button label="8" />
          <Button label="9" />
          <Button label="x" className="color" />
          <Button label="4" />
          <Button label="5" />
          <Button label="6" />
          <Button label="-" className="color" />
          <Button label="1" />
          <Button label="2" />
          <Button label="3" />
          <Button label="+" className="color" />
          <Button label="0" className="zero" />
          <Button label="." />
          <Button label="=" className="btn color" />
        </section>
      </main>
    </div>
  );
};

export default Calculator;
