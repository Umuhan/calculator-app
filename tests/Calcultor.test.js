import React from "react";
import renderer from "react-test-renderer";
import Calculator from "../src/components/Calculator";

test("Calculator snapshot test", () => {
  const component = renderer.create(<Calculator />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
