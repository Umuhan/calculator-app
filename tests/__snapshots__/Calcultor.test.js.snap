// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1
    className="math-magic"
    style={
      Object {
        "textAlign": "left",
      }
    }
  >
    Math Calculator
  </h1>
  <h4
    className="heading"
  >
    Lets do some math!
  </h4>
  <main
    className="container"
  >
    <section
      className="calculator"
    >
      <header
        className="header calc"
      >
        0
         
         
      </header>
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="color"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="color"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="color"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="color"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn color"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </section>
  </main>
</div>
`;
